### ISSUE TITLE
Script (web services) not launching with PHP8 in Archlinux( /Manjaro Linux) Install

### ISSUE THREAD

**Issue by rod2ik** [FOGProject/fogproject#447]
Hello,

Thanks for this wonderful software.

I am having problems to install with PHP 8 and on Manjaro Linux (not sure what is causing the problem yet). It should globally be similar to ArchLinux, I would say.

Could you please be of any help?

I have a request: Could you please develop the functionnalty for PHP8/Manjaro Linux, or  fix the installation bug for PHP8/Manjaro Linux? or at least help me get through the install process (I'm stuck on the activation of the web services).

(Following your install notes for archlinux), I just had to add a line `export PATH=$PATH:/usr/sbin` in my `.bashrc` root file to launch the install script On Manjaro Linux.

But the script fails to / does not activate the web services anymore, since PHP8. May be the same problem exists in the ArchLinux install with PHP8 ?

May be/perhaps/probably (?) because the following lines have changed (between PHP7 and PHP8) in the `/etc/httpd/conf/httpd.conf` apache config file, so as apache to run with PHP8 on Archlinux (and Manjaro Linux):

```
# PHP8 compatible way to configure the apache conf file in ArhcLinux/Manjaro Linux

# The following line has to be commented
#LoadModule mpm_event_module modules/mod_mpm_event.so

# this one must be uncommented
LoadModule mpm_prefork_module modules/mod_mpm_prefork.so

# Load php8 module
LoadModule php_module modules/libphp.so

# Uncomment the line
Include conf/extra/php_module.conf
```

May be you could guide me through what has to be changed/adapted?

Thanks very for any reply,

**Comment by Sebastian-Roth** [comment]
@rod2ik Interestingly we've had a post on Arch in the forums that was updated just recently: https://forums.fogproject.org/topic/15594/got-arch-manjaro-install-to-work

We are very happy to make this work on Manjaro and Arch but it'd be very handy to have exact information on what needs to be changed. Best if you can send in a pull request (against `dev-branch` not `master`) or post a `git diff` output for us to include.

> The following config lines have changed (between PHP7 and PHP8) in the /etc/httpd/conf/httpd.conf apache config file, to run apache with PHP8 on Archlinux (and Manjaro Linux).

We are talking about this part of the installer I guess: https://github.com/FOGProject/fogproject/blob/dev-branch/lib/common/functions.sh#L2107

If you're unable to send in a pull request of diff you might send us a default httpd.conf before adjustments.


> I had to add a line export PATH=$PATH:/usr/sbin in my .bashrc root file to launch the install script on Manjaro Linux: otherwise the script won't launch.

I know the check on `PATH` in the installer is not ideal to see if people use a proper root environment. If you have other ideas on how to make sure, I am happy to know about it. In the past we had people use `sudo ./installfog.sh` or `su` instead of `su -` and ran into strange problems. Therefore we added the check at some point.

**Comment by Sebastian-Roth** [comment]
@rod2ik Please help us with some more details.

**Comment by rod2ik** [comment]
Thanks for your reply. I will take some spare time to try to give you the best answer ASAP

**Comment by Sebastian-Roth** [comment]
@rod2ik Hope you had a good start to the new year. Any news on this issue?

**Comment by Sebastian-Roth** [comment]
I just got my act together and tried to find most references we have to FOG supporting Arch Linux.

https://forums.fogproject.org/topic/1697/install-fog-on-arch-linux
https://forums.fogproject.org/topic/6522/latest-git-fails-during-install-on-arch-linux

As well I am trying to reach out to all the people being involved with in the past: @dfear @mastacontrola @gfokkema @jonbesga @ase1590 @wayneworkman @rod2ik

Is anyone of you still using FOG on Arch Linux? And if yes, then is anyone of you keen and able to help update the code (mainly bash installer scripts)?

Might sound a bit harsh but Arch support will be removed from the code if we don't find anyone to work on this part. We won't purge github history so if someone comes along later we can still get things back without much effort. So if we don't hear back from anyone taking the lead within the next two weeks on this we'll remove Arch support.

**Comment by ase1590** [comment]
I've retired my FOG Arch server after a career change.

I'd say I agree, dropping arch support for the time being unless someone else wants to stop in would be the right move.

**Comment by Sebastian-Roth** [comment]
@ase1590 Thanks for your quick response.

I just figured I should ask in the forums as well before finally tossing it out: https://forums.fogproject.org/topic/15591/arch-manjaro-install-database-and-web-fails

So there is a last chance here we find someone willing to help on the ARCH support.

**Comment by mastacontrola** [comment]
Picked this up after four years. Splitting it, because the two halves have very different answers.

## The `export PATH=$PATH:/usr/sbin` half — fixed

Fixed on both branches: `40dbfffd4` (working-1.6), `c7c262806` (dev-branch).

@rod2ik your workaround is the thing that gave it away. The installer gated on root's environment by *inference* rather than by test:

```bash
which useradd >/dev/null 2>&1
if [[ $? -eq 1 || $(echo $PATH | grep -o "sbin" | wc -l) -lt 2 ]]; then
```

That second clause demands the literal string `sbin` appear **at least twice** in `$PATH`, standing in for "`/sbin` and `/usr/sbin` are both listed". Arch's root login PATH is:

```
/usr/local/sbin:/usr/local/bin:/usr/bin
```

`sbin` occurs **once**, so the installer refused to start and told you to load root's environment — which is exactly what you had already done. And appending `/usr/sbin` fixed it while adding no binaries whatsoever, because on Arch `/usr/sbin` is a symlink to `/usr/bin`. It only made the substring appear a second time. The check was measuring nothing.

@Sebastian-Roth — "the check on `PATH` is not ideal" was an understatement, and this is no longer Arch-only. Fedora has since finished the same `/usr/sbin` → `/usr/bin` merge and dropped sbin from root's PATH entirely:

```
# fedora:43, env -i bash -l
PATH=/root/.local/bin:/root/bin:/usr/local/bin:/usr/bin     (sbin x0)
```

So **the current installer refuses to run for root on Fedora 43 and Rawhide.** Same defect, live today, on a first-class supported platform.

The gate now adds the standard sbin directories to `PATH` when they exist and are not already listed, then simply checks the tool can be found. Old vs new, side by side under a true root login shell (`env -i bash -l`):

| image | sbin in PATH | old | new |
|---|---|---|---|
| archlinux | ×1 | REFUSED | allowed |
| debian:12 | ×3 | allowed | allowed |
| fedora:41 | ×2 | allowed | allowed |
| alpine:3.20 | ×3 | allowed | allowed |
| fedora:43 | ×0 | **REFUSED** | allowed |
| fedora:45 (rawhide) | ×0 | **REFUSED** | allowed |

It is strictly no more restrictive than before on every platform tested. (I deliberately did *not* widen it to `groupadd`/`usermod` — Alpine has neither, so that would have refused every Alpine install.)

You no longer need the `.bashrc` line on Arch.

## The PHP 8 / web services half — needs a maintainer decision, not a patch

Being straight about where this stands rather than leaving it open another four years:

- **On `working-1.6`, Arch is already gone.** There is no menu entry, no `*arch*` detection, and `osid=3` has been reassigned to Alpine. `lib/arch/config.sh` still sits in the tree but nothing references it — it is orphaned. The removal @Sebastian-Roth announced in 2022 effectively happened.
- **On `dev-branch` it is still selectable** (`3) Arch Linux`) and configures Apache for **mpm_event + mod_proxy_fcgi**, i.e. PHP-FPM. That is the *modern* arrangement and is PHP 8 compatible. The `mpm_prefork` + `LoadModule php_module modules/libphp.so` recipe in the original post is the older mod_php arrangement, which conflicts with event MPM — so applying it as written would fight the installer rather than fix it.
- Separately, `adduser` **does not exist on Arch at all** (only `useradd`), which `working-1.6` now depends on.

So the honest position is that nobody has run FOG on Arch in four years, nobody volunteered when asked, 1.6 has already dropped it, and there is no Arch box here to verify a blind Apache/PHP change against. Shipping an untested config rewrite for an unsupported platform would be worse than shipping nothing.

Leaving this open for @Tom-Elliott to call: either drop Arch from `dev-branch` too and tidy the orphaned `lib/arch/config.sh` on 1.6, or keep it if someone steps forward to test. The PATH fix above stands on its own either way and helps every distro.

**Comment by mastacontrola** [comment]
Arch is supported again, as **osid 4**, verified by installing FOG end to end on it. `db43cd8ab` on `working-1.6`.

Four years late, so a full accounting.

## The PATH half (fixed earlier, `40dbfffd4`)

Covered in my previous comment — the "sbin appears twice in $PATH" check was measuring nothing, and @rod2ik's `/usr/sbin` workaround proved it, since on Arch that path is a symlink to `/usr/bin`.

## The PHP 8 half — the proposed fix would not have worked

The recipe in the original post enables `mpm_prefork` and `LoadModule php_module modules/libphp.so`. That is **mod_php**, which forces the prefork MPM. FOG drives PHP through **php-fpm over mod_proxy_fcgi under the event MPM**, so applying that recipe would have fought the installer rather than fixed it. PHP 8 was never really the obstacle here.

`php-apache` is deliberately not in the package list for that reason.

## What was actually broken

Each of these cost a build cycle in an Arch container:

| | |
|---|---|
| `pacman -Sy` | Arch does not support partial upgrades. Refreshing the databases and then installing gave a php linked against a newer glibc than the one on disk — `libm.so.6: version GLIBC_2.44 not found`, a php that will not run at all. Now `-Syu`. |
| package ≠ service | The package is `apache`, the unit is `httpd.service`. Arch is the only distro where these differ, and `$webserver` was being used as both. |
| MariaDB | Arch leaves the data directory to the admin; its package runs no post-install setup. Without `mariadb-install-db` the service aborts on `Table 'mysql.db' doesn't exist`. 1.5 had this under osid 3 back when 3 meant Arch — it was dropped rather than renumbered when 3 became Alpine. |
| `pod2man` | Lives in `/usr/bin/core_perl`, put on PATH only by `/etc/profile.d/perlbin.sh`. Fails very late — the UDPCast build, after everything else has succeeded. |
| `adduser` | Does not exist on Arch at all. `configureUsers` now falls back to `useradd`, while Debian and RedHat keep the exact path they had. |
| `httpd.conf` | Ships with nearly every module commented out. |
| dropped packages | `mod_fastcgi`, `php-ssh2` and `xinetd` are gone from the repositories and would have failed the transaction. |

## Why 4 and not 3

**Alpine is not an Arch derivative** — musl/busybox/apk/OpenRC against glibc/coreutils/pacman/systemd. They share nothing but a couple of letters, so they get separate ids. Arch is 4, Alpine stays 3.

Since **3 meant Arch on the 1.5 line**, an Arch box upgrading from 1.5 carries `osid=3` and would otherwise have been silently configured as Alpine. That case now checks what the machine actually is and moves it to 4.

## Verified

On Arch: osid 4 detected and persisted, all 30 packages installed, `fogproject` created, MariaDB initialised and started, httpd + php-fpm active, schema updated, UDPCast built, TFTP and vsftpd up, **all eight FOG daemons active**, and `/fog/management/index.php` returning 200 with the login page. All eleven PHP extensions FOG needs are loaded.

Regression-checked on Debian 12 — unchanged, still creates the account with `adduser --system`, web UI 200.

## Honest limits

- The install stops at NFS in a container (kernel nfsd dependency), same as every other distro tested this way. **PXE booting and NFS serving have not been exercised** — this is "installs and runs", not "images machines". Real-hardware confirmation would be very welcome.
- `-S/--force-https` still fails on Arch, but not for an Arch reason: the vendored iPXE source will not assemble against current binutils, which treats its `.note.GNU-stack` warnings as errors. That is a shared-source problem affecting any distro with new-enough binutils, and I have left it alone rather than patch vendored source as a side effect of this.

@Sebastian-Roth — the "nobody has an Arch box" problem that closed this out in 2022 has genuinely gone away; the whole thing was developed and verified in throwaway containers.

`dev-branch` is unchanged and Arch remains unsupported there, by maintainer decision.
